0.8.1-Release-notes (#49)
percona/pcsm-docs: docs/release-notes/0.8.1.md

# {{ pcsm.full_name }} 0.8.1 ({{ date.v0_8_1 }})

We announce the release of Percona ClusterSync for MongoDB (PCSM) 0.8.1. This patch focuses on stability improvements, addressing a critical bug that could abort change-stream replication under heavy update workloads. With this fix, replication is more resilient and migrations complete reliably even when the source produces large, complex update events.

[Get started with {{pcsm.short}}](../installation.md){.md-button}

## Changelog

### Bugs

> - [PCSM-305](https://perconadev.atlassian.net/browse/PCSM-305): Fixed an issue where change-stream replication could fail on the target when source updates combined a truncation of a large nested array with indexed writes inside that same array and sibling-field updates. PCSM now splits update operations that conflict with an array truncation into separate follow-up writes and sizes bulk batches by aggregate wire bytes, keeping every individual update and every bulk command safely below MongoDB's 125 MB limit.

Author: @sandraromanchenko Could you please check this comment.

> - [PCSM-305](https://perconadev.atlassian.net/browse/PCSM-305): Fixed an issue where change-stream replication could fail on the target when source updates combined a truncation of a large nested array with indexed writes inside that same array and sibling-field updates. PCSM now splits update operations that conflict with an array truncation into separate follow-up writes and sizes bulk batches by aggregate wire bytes, keeping every individual update and every bulk command safely below MongoDB's 125 MB limit.

Reviewer: @rasika-chivate, numbers are correct, if wording is confusing change it to "keeping every individual update and every bulk command safely below **internal 125 MB BufBuilder buffer limit**"

> - [PCSM-305](https://perconadev.atlassian.net/browse/PCSM-305): Fixed an issue where change-stream replication could fail on the target when source updates combined a truncation of a large nested array with indexed writes inside that same array and sibling-field updates. PCSM now splits update operations that conflict with an array truncation into separate follow-up writes and sizes bulk batches by aggregate wire bytes, keeping every individual update and every bulk command safely below MongoDB's 125 MB limit.

Author: Fixed. @sandraromanchenko Pls check.

# {{ pcsm.full_name }} 0.8.1 ({{ date.v0_8_1 }})

> We announce the release of Percona ClusterSync for MongoDB (PCSM) 0.8.1. This patch focuses on stability improvements, addressing a critical bug that could cause MongoDB nodes to crash under heavy replication loads. With this fix, replication is now more resilient, ensuring smoother migrations.

Reviewer: ```suggestion
We announce the release of Percona ClusterSync for MongoDB (PCSM) 0.8.1. This patch focuses on stability improvements, addressing a critical bug that could abort change-stream replication under heavy update workloads. With this fix, replication is more resilient and migrations complete reliably even when the source produces large, complex update events.
```

[Get started with {{pcsm.short}}](../installation.md){.md-button}

## Changelog

### Bugs

> - [PCSM-305](https://perconadev.atlassian.net/browse/PCSM-305): We have fixed an issue where heavy replication workloads caused MongoDB nodes to crash with this error:

Reviewer: ```suggestion
- [PCSM-305](https://perconadev.atlassian.net/browse/PCSM-305): Fixed an issue where change-stream replication could fail on the target when source updates combined a truncation of a large nested array with indexed writes inside that same array and sibling-field updates. PCSM now splits update operations that conflict with an array truncation into separate follow-up writes and sizes bulk batches by aggregate wire bytes, keeping every individual update and every bulk command safely below MongoDB's 125 MB limit.
```

    ```sh
    Plan executor error during update :: caused by :: BufBuilder attempted to grow() to 131072004 bytes, past the 125MB limit.
    ```

    This crash led to write contention, timeouts, and stalled migrations, especially when moving large collections from Atlas. 

> 

Reviewer: ```suggestion
```
